collect_app/src/test/java/org/odk/collect/android/formentry/audit/FormSaveViewModelTest.java

```diff
@@ -67,39 +66,36 @@ public void setup() {
     }
 
     @Test
-    public void saveForm_returnsNewSaveResult_inSavingState() {
-        LiveData<FormSaveViewModel.SaveResult> saveResult1 = viewModel.saveForm(Uri.parse("file://form"), true, "", false);
-        assertThat(saveResult1.getValue().getState(), equalTo(SAVING));
-
-        LiveData<FormSaveViewModel.SaveResult> saveResult2 = viewModel.saveForm(Uri.parse("file://form"), true, "", false);
-        assertThat(saveResult2, not(equalTo(saveResult1)));
+    public void saveForm_returnsSaveResult_inSavingState() {
+        viewModel.saveForm(Uri.parse("file://form"), true, "", false);
+        FormSaveViewModel.SaveResult saveResult1 = viewModel.getSavedResult().getValue();
+        assertThat(saveResult1.getState(), equalTo(SAVING));
     }
 
     @Test
     public void saveForm_wontRunMultipleSavesAtOnce() {
-        LiveData<FormSaveViewModel.SaveResult> saveResult1 = viewModel.saveForm(Uri.parse("file://form"), true, "", false);
-        LiveData<FormSaveViewModel.SaveResult> saveResult2 = viewModel.saveForm(Uri.parse("file://form"), true, "", false);
-
-        assertThat(saveResult1.getValue().getState(), equalTo(SAVING));
-        assertThat(saveResult2.getValue().getState(), equalTo(ALREADY_SAVING));
-        assertThat(Robolectric.getBackgroundThreadScheduler().size(), equalTo(1));
+        viewModel.saveForm(Uri.parse("file://form"), true, "", false);
+        FormSaveViewModel.SaveResult saveResult1 = viewModel.getSavedResult().getValue();
+        assertThat(saveResult1.getState(), equalTo(SAVING));
 
-        whenFormSaverFinishes(SaveFormToDisk.SAVED);
-        assertThat(saveResult1.getValue().getState(), equalTo(SAVED));
-        assertThat(saveResult2.getValue().getState(), equalTo(ALREADY_SAVING));
+        viewModel.saveForm(Uri.parse("file://form"), true, "", false);
+        FormSaveViewModel.SaveResult saveResult2 = viewModel.getSavedResult().getValue();
+        assertThat(saveResult2.getState(), equalTo(ALREADY_SAVING));
```

Comment:
Sorry I don't think what I meant was clear here: I think we should get rid of the `ALREADY_SAVING` state entirely. It's unlikely that it would happen but if it did and then the user rotated the `LiveData` would be in the `ALREADY_SAVING` state rather than `SAVING` when the observation is setup again.

I think we can just return in `saveForm` if a save is in progress.

Reply:
So you mean to remove the last three lines at all? Then the test would be equal to the one above.

Reply:
Yeah I don't think we need this test because my feeling is that we should remove the line that this drives out in `FormSaveViewModel` (`saveResult.setValue(new SaveResult(SaveResult.State.ALREADY_SAVING, null))`). Really we should change it so that the test checks we don't call `formSaver#save` again - thinking about it the original test should really have had that check as well.